### ISSUE TITLE
Erroneous type error for addition of lists containing tuples containing strings

### ISSUE THREAD

**Issue by orestisfl** [microsoft/pyright#2754]
**Describe the bug**

In the following MWE pyright throws an error:

```python
a = [("a",)]
a + [("b",)]
```

Also same for (notice the `=`):

```python
a = [("a",)]
a += [("b",)]
```

The following example is (correctly) without an error:

```python
a = [("a",)]
a += [("b",), ("c",)] # weirdly, there is an error for this example if you remove the =
```

**To Reproduce**

```shell
$ npx -p pyright@1.1.199 pyright a.py
No configuration file found.
No pyproject.toml file found.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/a.py
  /tmp/a.py:2:1 - error: Operator "+" not supported for types "list[tuple[Literal['a']]]" and "list[tuple[Literal['b']]]"
    Operator "+" not supported for types "list[tuple[Literal['a']]]" and "list[tuple[Literal['b']]]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 1.155sec
```

**Expected behavior**

No error

**VS Code extension or command-line**

Latest pyright cli 1.1.199

**Comment by erictraut** [comment]
This is "as designed" behavior. The issue is that you haven't provided a type declaration for these variables, so pyright falls back on its type inference rules, which are documented [here](https://github.com/microsoft/pyright/blob/main/docs/type-inference.md).

The recommended approach in this case is to provide an appropriate type declaration.

```python
a: list[tuple[str]] = [("a",)]
a + [("b",)]
```

or

```python
a: list[tuple[Literal["a", "b"]]] = [("a",)]
a + [("b",)]
```

or

```python
a: list[tuple[str, ...]] = [("a",)]
a + [("b",)]
```

Your third example should generate an error. This operation is guded by the typeshed stub `builtins.pyi`, which defines the `__iadd__` method for the `list` class as follows:
```python
    def __iadd__(self: _S, __x: Iterable[_T]) -> _S: ...
```
I'll look into this case further because I think pyright should be generating an error in this case.

**Comment by erictraut** [comment]
The third case is also correct given the standard (non-strict) type inference rules used by pyright. If you enable strict list inference mode by adding a `# pyright: strictListInference=true` to the top of the file (or `# pyright: strict`), you'll see that the third case also emits an error. When strict list inference is disabled, the inferred type of the list is `list[Unknown]`, which is why it doesn't emit an error in this case.

So all of this is working as expected.
